swift/ql/lib/codeql/swift/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -125,7 +125,9 @@ module Stmts {
       c instanceof SimpleCompletion
     }
 
-    predicate firstInner(ControlFlowElement first) { astFirst(ast.getFirstElement(), first) }
+    predicate firstInner(ControlFlowElement first) {
+      astFirst(ast.getFirstElement().getFullyUnresolved(), first)
+    }
 
     /** Gets the body of the i'th `defer` statement. */
     private BraceStmt getDeferStmtBody(int i) {
```

Comment:
I'm guessing the `getElement` in here does not need to be fully unresolved because `Stmt`s are never resolved, right? Still if we define `getElement` predicates here, I would just `s/ast.get(\w*)Element/this.get\1Element/` in the whole class, as having to think where to use one or the other becomes hard (and one never know if we ever start resolving other AST node kinds)

Reply:
Yeah, we can probably use `getFullyUnresolved` everywhere (with the exception of inside subexpressions of conversions since this would get us into an infinite loop).

Reply:
I think we can avoid the infinite loop if we stick to using `getImmediate` getters consistently

Reply:
Agreed

Reply:
Should we maybe just merge this PR as-is and then do your suggestion as a follow-up @redsun82 ?